tests/src/test/scala/whisk/core/containerpool/kubernetes/test/KubernetesClientTests.scala

```diff
@@ -0,0 +1,272 @@
+ * limitations under the License.
+ */
+
+package whisk.core.containerpool.kubernetes.test
+
+import java.time.LocalDateTime
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Concat, Flow, Framing, Sink, Source}
+import akka.util.ByteString
+
+import scala.concurrent.Await
+import scala.concurrent.ExecutionContext
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import spray.json._
+import org.junit.runner.RunWith
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.concurrent.Eventually
+import org.scalatest.FlatSpec
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.Matchers
+import org.scalatest.time.{Seconds, Span}
+import common.{StreamLogging, WskActorSystem}
+import okio.Buffer
+import whisk.common.LogMarker
+import whisk.common.LoggingMarkers.INVOKER_KUBECTL_CMD
+import whisk.common.TransactionId
+import whisk.core.containerpool.{ContainerAddress, ContainerId}
+import whisk.core.containerpool.kubernetes.{KubernetesApi, KubernetesClient, KubernetesRestLogSourceStage}
+import whisk.core.containerpool.docker.ProcessRunningException
+import whisk.core.containerpool.logging.LogLine
+
+import scala.collection.mutable
+import scala.util.Try
+@RunWith(classOf[JUnitRunner])
+class KubernetesClientTests
+    extends FlatSpec
+    with Matchers
+    with StreamLogging
+    with BeforeAndAfterEach
+    with Eventually
+    with WskActorSystem {
+
+  import KubernetesClientTests._
+
+  implicit val materializer: ActorMaterializer = ActorMaterializer()
+
+  /** Reads logs into memory and awaits them */
+  def awaitLogs(source: Source[ByteString, Any], timeout: FiniteDuration = 1000.milliseconds): Vector[LogLine] =
+    Await
+      .result(
+        source
+          .via(Framing.delimiter(ByteString("\n"), 1024))
+          .via(Flow[ByteString].map(l => Try(l.utf8String.parseJson.convertTo[LogLine]).getOrElse(LogLine("", "", ""))))
+          .filter(_ != LogLine("", "", ""))
+          .runWith(Sink.seq[LogLine]),
+        timeout)
+      .toVector
+
+  override def beforeEach = stream.reset()
+
+  implicit override val patienceConfig = PatienceConfig(timeout = scaled(Span(5, Seconds)))
+
+  implicit val transid = TransactionId.testing
+  val id = ContainerId("55db56ee082239428b27d3728b4dd324c09068458aad9825727d5bfc1bba6d52")
+
+  val commandTimeout = 500.milliseconds
+  def await[A](f: Future[A], timeout: FiniteDuration = commandTimeout) = Await.result(f, timeout)
+
+  val kubectlCommand = "kubectl"
+
+  /** Returns a KubernetesClient with a mocked result for 'executeProcess' and 'fetchHTTPLogs' */
+  def kubernetesClient(fixture: => Future[String]) = new KubernetesClient()(global) {
+    override def findKubectlCmd() = kubectlCommand
+    override def executeProcess(args: Seq[String], timeout: Duration)(implicit ec: ExecutionContext, as: ActorSystem) =
+      fixture
```

Comment:
fwiw, I don't think mocking out executeProcess is going to be a useful long-term tool for unit tests.  As we replace kubectl commands with calls to the kubernertes REST API, we lose executeProcess as a hook point.  For example, in a branch based off this PR, I've switched to doing pod creation via the REST API and as a result lost this hook point for all the "run" unit tests (so I disabled them).

Reply:
DISABLED THE TESTS WHAT!?!?! ;)
Yeah, @bwmcadams  had to set up dummy `Source` objects to validate the log parsing logic. I guess we'll need to figure out something similar as we convert the other calls to the API.

Reply:
open for suggestions on how to unit test run when it becomes just a thin veneer over the kubernetesClient API.  Code is sitting here if you want a preview: https://github.com/dgrove-oss/openwhisk/tree/k8s-invokeragent-20180219 (I've cleaned up & consolidated my older branches into 3 logical commits when I rebased them onto the latest code from this PR).